<?xml version="1.0"?>

<launch>

  <!-- IMU -->
  <node pkg="imu" type="imu.py" name="imu" respawn="true">
  </node>

  <!-- MOTORS -->

  <arg name="test_mode" default="False"/>

  <node pkg="dc_motor_driver" type="motor_driver.py" name="motor_driver" respawn="true">
    <param name="test_mode" value="$(arg test_mode)" />
    <param name="param_path" value="$(find dc_motor_driver)/params/motor_params.yaml"/>
  </node>

  <!-- SONARS -->
  <!-- Front -->
  <node pkg="read_sonars" type="read_sonars.py" name="read_sonars_f" output="screen"> 
    <param name="sonar_topic" value="sonar_front" />
    <param name="sonar_echo" value="20" />
    <param name="sonar_trigger" value="21" /> 
  </node>
  <!-- Front  Left-->	
  <node pkg="read_sonars" type="read_sonars.py" name="read_sonars_fl" > 
    <param name="sonar_topic" value="sonar_front_left" />
    <param name="sonar_echo" value="16" />
    <param name="sonar_trigger" value="19" /> 
  </node>
  <!-- Front  Right-->
  <node pkg="read_sonars" type="read_sonars.py" name="read_sonars_fr" > 
    <param name="sonar_topic" value="sonar_front_right" />
    <param name="sonar_echo" value="12" />
    <param name="sonar_trigger" value="13" /> 
  </node>
  <!-- Left -->
  <node pkg="read_sonars" type="read_sonars.py" name="read_sonars_l" > 
    <param name="sonar_topic" value="sonar_left" />
    <param name="sonar_echo" value="6" />
    <param name="sonar_trigger" value="25" /> 
  </node>
  <!-- Right -->
  <node pkg="read_sonars" type="read_sonars.py" name="read_sonars_r" > 
    <param name="sonar_topic" value="sonar_right" />
    <param name="sonar_echo" value="5" />
    <param name="sonar_trigger" value="24" /> 
  </node>

  <!-- Localizer -->

  <node pkg="state_estimation" type="localizer_thread.py" name="localizer" output="screen">
  </node>

  <!-- Planner -->

  <node pkg="alpha_star" type="a_star.py" name="a_star" output="screen">
  </node>

</launch>
